src/endpoint.cpp

```diff
@@ -213,17 +218,68 @@ namespace oxen::quic
         return std::make_optional<ConnectionID>(vid.dcid, vid.dcidlen);
     }
 
+    Connection* Endpoint::accept_initial_connection(Packet& pkt, ConnectionID& dcid)
+    {
+        log::info(log_cat, "Accepting new connection...");
+
+        ngtcp2_pkt_hd hdr;
+
+        auto rv = ngtcp2_accept(&hdr, u8data(pkt.data), pkt.data.size());
+
+        if (rv < 0)  // catches all other possible ngtcp2 errors
+        {
+            log::warning(
+                    log_cat,
+                    "Warning: unexpected packet received, length={}, code={}, continuing...",
+                    pkt.data.size(),
+                    ngtcp2_strerror(rv));
+            return nullptr;
+        }
+        if (hdr.type == NGTCP2_PKT_0RTT)
+        {
+            log::error(
+                    log_cat,
+                    "Error: 0RTT is currently not utilized in this implementation; dropping "
+                    "packet");
+            return nullptr;
+        }
+        if (hdr.type == NGTCP2_PKT_INITIAL && hdr.tokenlen)
+        {
+            log::warning(log_cat, "Warning: Unexpected token in initial packet");
+            return nullptr;
+        }
+
+        for (;;)
+        {
+            if (auto [itr, res] = conns.emplace(ConnectionID::random(), nullptr); res)
+            {
+                std::promise<Connection*> p;
+                auto f = p.get_future();
+
+                net.call([this, pkt, &p, &hdr, it = itr]() {
+                    it->second = std::move(Connection::make_conn(
+                            *this, it->first, hdr.scid, pkt.path, handle, inbound_ctx->tls_creds, inbound_ctx->config, Direction::INBOUND, &hdr));
+
+                    p.set_value(it->second.get());
+                    return;
+                });
+
+                return f.get();
+            }
+        };
+    };
+
     void Endpoint::handle_conn_packet(Connection& conn, Packet& pkt)
```

Comment:
This seems misplaced: everything in here, I think, could be in a Connection method.